static/app/actionCreators/pageFilters.tsx

```diff
@@ -299,17 +313,34 @@ function updateParams(obj: PageFiltersUpdate, router?: Router, options?: Options
     return;
   }
 
-  if (options?.save) {
-    const {organization} = OrganizationStore.getState();
-    const {selection} = PageFiltersStore.getState();
-    const orgSlug = organization?.slug ?? null;
+  const routerAction = options?.replace ? router.replace : router.push;
+
+  routerAction({pathname: router.location.pathname, query: newQuery});
+}
 
-    setPageFiltersStorage(orgSlug, selection, newQuery);
+/**
+ * Save the current page filters to local storage
+ */
+async function persistPageFilters(options?: Options) {
+  if (!options?.save) {
+    return;
   }
 
-  const routerAction = options?.replace ? router.replace : router.push;
+  // XXX(epurkhiser): Since this is called immediately after updating the
+  // store, wait for a tick since stores are not updated fully synchronously..
+  // a bit goofy, but it works fine.
+  await new Promise(resolve => setTimeout(resolve, 0));
```

Comment:
This is definitely questionable. Open to thoughts on other ways to handle this.

Maybe persistence should be triggered in the store itself? I think I would kind of prefer not 🤔

Reply:
Doing persistence changes in the store would allow you to skip this tick skip :smile:. I think the store might be a better place for persistence as it is supposed to be the source of state. Couldn't we currently get into a state where the the store and local storage have different state?

Reply:
Yeah, there's actually a number of places right now where we don't persist changes to the state. Anywhere that we don't pass `save: true` actually, which I think right now is _only_ in the selector components themselves.

Reply:
This will be changing in the pretty immediate future as well, since soon we'll _only_ be persisting when pinning filters, at which point I think it would be easier to move this into the store probably.

So for now, I think this is OK and we can revisit later